### PR TITLE
don't fail on unavailble module nupkg (for when publishing release with assets in build.yml)

### PR DESCRIPTION
**Pull request by gaelcolas**
This is to enable the scenario where we don't produce a PowerShell module,
but for instance Chocolatey Package(s), that can be different from the Project Name.

Say you have several Choco packages in a repo, or you have several Azure Policy Guest Configuration Packages, you'd add the list of assets you want to publish in GitHub by listing them in the Build.Yaml (with a * for the version).
```yaml
####################################################
#               GitHub Configuration               #
####################################################
GitHubConfig:
  ReleaseAssets:
    - 'output/CHANGELOG.md'
    - 'output/MyPackage1*.nupkg
    - 'output/MyPackage2*.nupkg
  GitHubFilesToAdd:
    - 'CHANGELOG.md'
  GitHubConfigUserName: blah
  GitHubConfigUserEmail: blah@foobar.com
  UpdateChangelogOnPrerelease: false
```

### CODE REVIEW

File: .build/tasks/New-Release.GitHub.build.ps1 [hunk 1]
```diff
@@ -109,10 +111,7 @@ task Publish_release_to_GitHub -if ($GitHubToken -and (Get-Module -Name PowerShe
             ErrorAction    = 'Stop'
         }
 
-        $displayGHReleaseParams = $getGHReleaseParams.Clone()
```
Comment (johlju) [inline .build/tasks/New-Release.GitHub.build.ps1:112]: Why was this removed? This made sure we don't show the token.

File: .build/tasks/New-Release.GitHub.build.ps1 [hunk 2]
```diff
@@ -65,16 +64,19 @@ task Publish_release_to_GitHub -if ($GitHubToken -and (Get-Module -Name PowerShe
...
+    $packagedProjectNupkg = Join-Path -Path $OutputDirectory -ChildPath "$ProjectName.$moduleVersion.nupkg"
+    $PackageToRelease = Get-ChildItem -Path $packagedProjectNupkg -ErrorAction Ignore
+    # If the Project nupkg is not found, don't fail. You can still create a release and specify the
+    # assets in the build.yml (i.e. Chocolatey packages or Azure Policy Guest Config Packages)
     $ReleaseTag = "v$ModuleVersion"
 
     Write-Build DarkGray "About to release '$PackageToRelease' with tag and release name '$ReleaseTag'"
     $remoteURL = git remote get-url origin
 
     if ($remoteURL -notMatch 'github')
```
Comment (johlju) [inline .build/tasks/New-Release.GitHub.build.ps1:77]: Not sure this is needed since the GitHubToken need to be set to even run the task? 🤔

**Review by johlju — CHANGES_REQUESTED** [review]

File: .build/tasks/New-Release.GitHub.build.ps1 (hunk 2 above)
Comment (gaelcolas) [reply .build/tasks/New-Release.GitHub.build.ps1:77]: Good point.

**Comment by gaelcolas** [comment]
Hopefully that's all fixed.

**Review by johlju — APPROVED** [review]

**Comment by johlju** [comment]
Seems this might be an issue if it was expected to create a nupkg but none was created. Hopefully it fails further down the pipeline in that case. 🙂

**Comment by johlju** [comment]
@gaelcolas leave it fior you to merge and release.

**Comment by gaelcolas** [comment]
> Seems this might be an issue if it was expected to create a nupkg but none was created. Hopefully it fails further down the pipeline in that case. 🙂

I thought of that, but what should fail is the task that creates the Nupkg.
Usually, even if the nupkg is not created for some reason, I'd rather have the release created so I can add the asset later, but that might be just personal preference.
Anyhow, this task should just be about creating a GH release with what's available, as you said, errors should happen elsewhere.
